### PR TITLE
Remove duplicate code

### PR DESCRIPTION
**Pull request by cmpute**
Minor changes

### CODE REVIEW

**Comment by lvhan028** [comment]
May resolve the linting error.  The following command can help verify and fix the linting issues.
```
pip install pre-commit
cd lmdeploy
pre-commit install
pre-commit run --all-files
```

File: lmdeploy/serve/openai/api_client.py [hunk 1]
```diff
@@ -7,11 +7,15 @@
...
+    if not response.ok:
+        logger = get_logger('lmdeploy')
+        logger.error(f'Failed to get the model list: {api_url} returns {response.status_code}')
+        return None
     if hasattr(response, 'text'):
-        model_list = json.loads(response.text)
+        model_list = response.json()
         model_list = model_list.pop('data', [])
         return [item['id'] for item in model_list]
     return None
```
Comment (AllentDan) [inline lmdeploy/serve/openai/api_client.py:21]: there are two `return None`. May make it cleaner.

Comment (cmpute) [reply lmdeploy/serve/openai/api_client.py]: Fixed

**Comment by cmpute** [comment]
Please review once more

**Comment by lvhan028** [comment]
There is linting issue. Please fix it according the the guide before

**Comment by cmpute** [comment]
image

Some yapf changes are not due to my PR. I won't fix those.

**Review by AllentDan — APPROVED** [review]
LGTM

**Pull request merged**